File: src/tests/fidl/source_compatibility/union-strict-flexible/go/step_00_before.go
// Copyright 2021 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package main

import (
	lib "fidl/fidl/test/unionstrictflexible"
	"fmt"
)

// [START contents]
func useUnion(value lib.JsonValue) {
	switch value.Which() {
	case lib.JsonValueIntValue:
		fmt.Printf("int value: %d\n", value.IntValue)
	case lib.JsonValueStringValue:
		fmt.Printf("string value: %s\n", value.StringValue)
	default:
		fmt.Println("unknown tag")
	}
}

// [END contents]

func main() {}

File: src/tests/fidl/source_compatibility/table-member-remove/llcpp/step_00_before.cc
// Copyright 2021 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include <fidl/test/tablememberremove/llcpp/fidl.h>  // nogncheck
namespace fidl_test = llcpp::fidl::test::tablememberremove;

// [START contents]
void use_table(const fidl_test::Profile& profile) {
  if (profile.has_timezone()) {
    printf("timezone: %s", profile.timezone().data());
  }
  if (profile.has_temperature_unit()) {
    printf("preferred unit: %s", profile.temperature_unit().data());
  }
  if (profile.has_dark_mode()) {
    printf("dark mode on: %s", profile.dark_mode() ? "true" : "false");
  }
}
// [END contents]

int main(int argc, const char** argv) { return 0; }
